CONTRIBUTING.md

```diff
@@ -0,0 +1,76 @@
+# How to contribute
+
+Thank you for your interest in contributing to the FeedHenry project. We want
+keep this process as easy as possible so we've outlined a few guidelines below.
+For more information about the FeedHenry project, visit
+[our website](http://feedhenry.org/).
+
+## Asking for help
+
+Whether you're contributing a new feature or bug fix, or simply submitting a
+ticket, the FeedHenry team is available for technical advice or feedback.
+You can reach us at #feedhenry and #feedhenry-raincatcher on [Freenode IRC](https://freenode.net/) or the
+[feedhenry-dev list](https://www.redhat.com/mailman/listinfo/feedhenry-dev)
+-- both are actively monitored.
+
+## Getting started
+
+* Make sure you have a [JIRA account](https://issues.jboss.org)
+* Make sure you have a [GitHub account](https://github.com/signup/free)
+* Submit a ticket for your issue to the
+[RainCatcher project](https://issues.jboss.org/projects/RAINCATCH/), assuming one does
+not already exist.
+  * Clearly describe the issue including steps to reproduce when it is a bug.
+  * Make sure you fill in the earliest version that you know has the issue.
+* Fork the repository on GitHub.
+
+## Making changes
+
+* Create a topic branch from where you want to base your work.
+  * This is usually the master branch.
+  * To quickly create a topic branch based on master; `git checkout -b
+    <branch name> master`. By convention we typically include the JIRA issue
+    key in the branch name, e.g. `RAINCATCH-1234-my-feature`.
+  * Please avoid working directly on the `master` branch.
+* Make commits of logical units.
+* Prepend your commit messages with a JIRA ticket number, e.g. "RAINCATCH-1234: Fix
+  spelling mistake in README."
+* Follow the coding style in use.
+* Check for unnecessary whitespace with `git diff --check` before committing.
+* Make sure you have added the necessary tests for your changes.
+* Run _all_ the tests to assure nothing else was accidentally broken.
+
+## Submitting changes
+
+* Push your changes to a topic branch in your fork of the repository.
+* Submit a pull request to the repository in the [RainCatcher GitHub organization]
+  (https://github.com/feedhenry-raincatcher) and choose branch you want to patch
+  (usually master).
+  * Advanced users may want to install the [GitHub CLI](https://hub.github.com/)
+    and use the `hub pull-request` command.
+* Update your JIRA ticket to mark that you have submitted code and are ready
+for it to be reviewed (Status: Dev Complete).
```

Comment:
@JameelB yeah like wojtek said I only lazily copied over the CONTRIBUTING guide that we had previously on a couple of the raincatcher modules.

@wtrocki I've also seen that some projects move the details around code structure and dev tools setup to this file instead of on the top readme, do you think we should too?